pkg/unpack/unpacker.go

```diff
@@ -240,7 +240,14 @@ func (u *Unpacker) unpack(
 	// TODO: Support multiple unpacks rather than just first match
 	var unpack *Platform
 
-	imgPlatform := platforms.Normalize(ocispec.Platform{OS: i.OS, Architecture: i.Architecture})
+	imgPlatform := platforms.Normalize(ocispec.Platform{
+		OS:           i.OS,
+		Architecture: i.Architecture,
+		Variant:      i.Variant,
+
+		OSVersion:  i.OSVersion,
+		OSFeatures: i.OSFeatures,
+	})
```

Comment:
We should consider updating the ocispec.Image to embed the ocispec.Platform 🤔 ; 

```go
// Image is the JSON structure which describes some basic information about the image.
// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
type Image struct {
	// Created is the combined date and time at which the image was created, formatted as defined by RFC 3339, section 5.6.
	Created *time.Time `json:"created,omitempty"`

	// Author defines the name and/or email address of the person or entity which created and is responsible for maintaining the image.
	Author string `json:"author,omitempty"`

	Platform

	// Config defines the execution parameters which should be used as a base when running a container using the image.
	Config ImageConfig `json:"config,omitempty"`

	// RootFS references the layer content addresses used by the image.
	RootFS RootFS `json:"rootfs"`

	// History describes the history of each layer.
	History []History `json:"history,omitempty"`
}
```


Then all this copying wouldn't be needed, and we could just do

```suggestion
	imgPlatform := platforms.Normalize(i.Platform)
```

Reply:
Violent agreement! :sweat_smile:

> There's probably a pretty decent argument to be made for the OCI's `Image` type to embed `Platform` directly (which would make this more straightforward) and/or to offer a helper function that appropriately extracts a `Platform` object _from_ an `Image` object, but I didn't want the perfect to be the enemy of the good, and this seemed like a good place to start.

:eyes: :heart: :smile:

Reply:
It would also be technically _correct_ in that case to swap back to using `Image` in all these cases, but it feels like a huge amount of overkill to process the entire `Image` object, extract all the data, and then throw away all but at most five fields of it :sweat_smile:

Reply:
Oh! Missed your last comments; I was updating my fork of the image-spec; https://github.com/opencontainers/image-spec/pull/949

😂

I agree that just marshaling/unmarshaling the platform makes a lot of sense; but indeed more "correct" if it's actually embedded, which ties them together to be the same thing.

Reply:
Given https://github.com/opencontainers/image-spec/pull/949 is merged (which further legitimizes this PR's implementation), is this ready for re-review or do we want to do https://github.com/containerd/containerd/pull/7382 first and rebase this on top of that? :bow: